friends/src/App.js

```diff
@@ -1,11 +1,84 @@
 import React, { Component } from 'react';
+import axios from 'axios';
 import './App.css';
-import Friends from 'Friends';
+import Friends from './components/Friends';
 class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      friends: [],
+      id: '',
+      name: '',
+      age: '',
+      email: '',
+    }
+  }
+
+  componentDidMount() {
+    axios
+      .get('http://localhost:5000/friends')
+      .then(response => {
+        console.log(response.data);
+        this.setState({ friends: response.data })
+      })
+  }
+
+  name = (event) => {
+    const newName = event.target.value;
+    this.setState({
+      name: newName
+    })
+  }
+  age = (event) => {
+    const newAge = event.target.value;
+    this.setState({
+      age: newAge
+    })
+  }
+  email = (event) => {
+    const newEmail = event.target.value;
+    this.setState({
+      email: newEmail
+    })
+  }
+  addFriend = (event) => {
+    event.preventDefault();
+    let newId = this.state.friends.length + 1;
+    this.setState({
+      id: newId
+    })
+
+    const newFriend = {
+      id: this.state.id,
+      name: this.state.name,
+      age: this.state.age,
+      email: this.state.email,
+    }
+
+    axios
+      .post('http://localhost:5000/friends', newFriend)
+      .then(response => {
+        this.setState({
+          friends: response.data,
+        })
+          
+      })
+      .catch(error => { console.log('error!') })
```

Comment:
Great job including the catch. I would not print out error as a string. I would print the variable error so that you can see what the request response was.